client/app/visualizations/chart/Renderer/PlotlyChart.jsx

```diff
@@ -1,42 +1,65 @@
 import { isArray, isObject } from "lodash";
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useContext } from "react";
+import { ErrorBoundaryContext } from "@/components/ErrorBoundary";
 import { RendererPropTypes } from "@/visualizations";
 import resizeObserver from "@/services/resizeObserver";
 
 import getChartData from "../getChartData";
 import { Plotly, prepareData, prepareLayout, updateData, applyLayoutFixes } from "../plotly";
 
+function catchErrors(func, errorHandler) {
+  return (...args) => {
+    try {
+      return func(...args);
+    } catch (error) {
+      errorHandler.handleError(error);
```

Comment:
What is this part for? Overall it seemed it only needed a `<ErrorBoundary>` component as a fallback in the tree (no need to use context for example), so was there any issues or there are any gains when you add this part?

Reply:
Error boundary will catch only errors that happened during rendering; this part is needed to catch async errors as well